src/driving/drifty_car.py

```diff
@@ -0,0 +1,230 @@
+        backwards. Excludes any lateral movement of the car.
+        """
+
+        # Compute how much the wheels are pushing the car forward due to throttle
+        if facing_speed > self.max_engine_speed:
+            forward_acceleration = (0, 0)
+        else:
+            # Sample a different curve when in "drift mode"
+            curve = (
+                self.forward_drifting_acceleration
+                if drifting
+                else self.forward_acceleration
+            )
+            # Maximum acceleration power, iff the player is fully holding the accelerate axis
+            max_forward_acceleration = curve.sample(facing_speed_percentage_of_max)
+            forward_acceleration = scale_vec(
+                facing,
+                self.input.accelerate_axis.value * max_forward_acceleration,
+            )
+
+        # Compute how much the wheels are pushing the car backward due to throttle
+        if facing_speed < -self.max_engine_speed:
+            reverse_acceleration = (0, 0)
+        else:
+            curve = (
+                self.reverse_drifting_acceleration
+                if drifting
+                else self.reverse_acceleration
+            )
+            max_reverse_acceleration = curve.sample(-facing_speed_percentage_of_max)
+            reverse_acceleration = scale_vec(
+                facing,
+                -self.input.brake_axis.value * max_reverse_acceleration,
+            )
+
+        # Compute automatic braking force applied when neither gas nor reverse
+        # is pressed
+        if braking and not drifting:
+            braking_power = self.braking_acceleration.sample(abs(facing_speed))
+            # Prevent braking friction from overshooting and causing the car to move in the
+            # opposite direction
+            # We must factor delta-time because we're clamping based on the net change in
+            # acceleration applied this frame, which is delta-time-d later.
+            braking_power = clamp(braking_power, 0, abs(facing_speed) / delta_time)
+            if facing_speed > 0:
+                braking_power = -braking_power
+            braking_acceleration = scale_vec(facing, braking_power)
+        else:
+            braking_acceleration = (0.0, 0.0)
+
+        # Compute friction of the wheels resisting the car sliding laterally
+        if drifting:
+            chosen_friction = 0
+        elif absolute_lateral_speed > self.lateral_velocity_kinetic_friction_threshold:
+            chosen_friction = self.kinetic_friction
+        else:
+            chosen_friction = self.static_friction
+        lateral_friction_acceleration = scale_vec(lateral_velocity, chosen_friction)
+
+        # Sum up net accelerations
+        acceleration = add_vec(forward_acceleration, reverse_acceleration)
+        acceleration = add_vec(acceleration, lateral_friction_acceleration)
+        acceleration = add_vec(acceleration, braking_acceleration)
```

Comment:
Note to self: follow up w/Dan after review, consider renaming `braking_acceleration`.

Reply:
Note to self: use the varargs form of `add_vec`